### PR TITLE
Adding External Scopes properly based on ARM Suggestion

### PR DESCRIPTION
**Pull request by parisa-naeimi**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/parisa-naeimi/azure-rest-api-specs/dev-cost-management-Microsoft.CostManagement-2019-11-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by parisa-naeimi** [comment]
@ChenTanyi  could you please take a look at this one and also can somebody from ARM team checks this one? should be straightforward?

**Comment by ChenTanyi** [comment]
@parisa-naeimi API change need ARM review first. SDK will be ok when all ci pass and no readme need to be added.

**Comment by ms-premp** [comment]
@chentanyi
there is no API change. The routes we had in swagger needed correction. It was wrong before.

**Comment by ChenTanyi** [comment]
@ms-premp All change for non examples json will be considered as API change. ARM will review it base on the label I add before. If the pr is emergent, you can send email to `armpm@microsoft.com` or `armapireview@microsoft.com`

**Comment by parisa-naeimi** [comment]
> @ms-premp All change for non examples json will be considered as API change. ARM will review it base on the label I add before. If the pr is emergent, you can send email to `armpm@microsoft.com` or `armapireview@microsoft.com`

Thank you for providing these email addresses, I sent them an email to review this PR.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json [hunk 1]
```diff
@@ -1473,6 +1606,32 @@
...
+        "externalSubscriptions",
+        "externalBillingAccounts"
+      ],
+      "x-ms-enum": {
+        "name": "externalCloudProviderType",
+        "modelAsString": true
+      },
+      "description": "The external cloud provider type associated with dimension/query operations. This includes 'externalSubscriptions' for linked account and 'externalBillingAccounts' for consolidated account.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
```
Comment (anthony-c-martin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:1625]: You shouldn't need to set `x-ms-skip-url-encoding` here

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json [hunk 2]
```diff
@@ -1473,6 +1606,32 @@
...
+    },
+    "externalCloudProviderIdParameter": {
+      "name": "externalCloudProviderId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "This can be '{externalSubscriptionId}' for linked account or '{externalBillingAccountId}' for consolidated account used with dimension/query operations.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
```
Comment (anthony-c-martin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:1634]: You shouldn't need to set `x-ms-skip-url-encoding` here

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json (hunk 1 above)
Comment (parisa-naeimi) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json]: Sure, I removed it. thank you for the quick review

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json (hunk 2 above)
Comment (parisa-naeimi) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json]: Removed if

**Comment by parisa-naeimi** [comment]
@ChenTanyi Arm approved my changes, could you please take a look and let us know if we need to take any action? Otherwise can you please merge this PR to master?

**Review by ChenTanyi — APPROVED** [review]

**Comment by parisa-naeimi** [comment]
@ChenTanyi  Thank you for approving and merging PR to master. Looking at cost management documents, it does not have my recent changes. How long does it take to see the impact of changes in document? I was wondering after merge to master, is there any other step needs to be done to show latest changes in document?

**Comment by ChenTanyi** [comment]
@parisa-naeimi I don't know about it. You can ask @akning-ms or `Ankun.Ning@microsoft.com`.

**Pull request merged**